benchmark/UnsortedAndSortedJoinBenchmark.cpp

```diff
@@ -0,0 +1,356 @@
+
+  // Because the sortThenJoinLambdaWrapper sorts tables IN PLACE, a and b are now sorted.
+  records->addToExistingTable("Sorted IdTables", 1, 1, hashJoinLambdaWrapper);
+  records->addToExistingTable("Sorted IdTables", 0, 1, joinLambdaWrapper);
+
+  // Benchmarking with overlapping IdTables.
+
+  // We make the tables overlapping and then randomly shuffle them.
+  createOverlapRandomly(&a, b, 10.0);
+
+  randomShuffle(a.idTable.begin(), a.idTable.end());
+  randomShuffle(b.idTable.begin(), b.idTable.end());
+
+  records->addToExistingTable("Unsorted IdTables", 1, 0, hashJoinLambdaWrapper);
+  records->addToExistingTable("Unsorted IdTables", 0, 0, sortThenJoinLambdaWrapper);
+
+  // Because the sortThenJoinLambdaWrapper sorts tables IN PLACE, a and b are now sorted.
+  records->addToExistingTable("Sorted IdTables", 1, 0, hashJoinLambdaWrapper);
+  records->addToExistingTable("Sorted IdTables", 0, 0, joinLambdaWrapper);
+}
+
+template<typename T1, typename T2, typename T3, typename T4>
+concept onlyFirstIsVector = std::is_same<T1, std::vector<size_t>>::value &&
+  std::is_same<T2, size_t>::value && std::is_same<T3, size_t>::value &&
+  std::is_same<T4, size_t>::value;
+
+/*
+ * @brief Create a benchmark table for join algorithm, with the given
+ *  parameters for the IdTables. The columns will be the algorithm and the
+ *  rows will be the parameter, you gave a list for.
+ *
+ * @tparam T1, T2, T3, T4 Exactly on of those muste be a std::vector<size_t>
+ *  and all others must be size_t. Typ inference should be able to handle it,
+ *  if you give the right function arguments.
+ *
+ * @param records The BenchmarkRecords, in which you want to create a new
+ *  benchmark table.
+ * @param overlap Should the tables used for the benchmark table have any
+ *  overlap concering the entries of their join columns?
+ * @param smallerTableSorted, biggerTableSorted Should the bigger/smaller table
+ *  be sorted by his join column before being joined? More specificly, some
+ *  join algorithm require one, or both, of the IdTables to be sorted. If this
+ *  argument is false, the time needed for sorting the required table will
+ *  added to the time of the join algorithm.
+ * @param ratioRows How many more rows than the smaller table should the
+ *  bigger table have? In more mathematical words: Number of rows of the
+ *  bigger table divided by the number of rows of the smaller table is equal
+ *  to ratioRows.
+ * @param smallerTableAmountRows How many rows should the smaller table have?
+ * @param smallerTableAmountColumns, biggerTableAmountColumns How many columns
+ *  should the bigger/smaller tables have?
+ */
+template<typename T1, typename T2, typename T3,typename T4>
+requires onlyFirstIsVector<T1, T2, T3, T4> || onlyFirstIsVector<T2, T1, T3, T4>
+  || onlyFirstIsVector<T3, T2, T1, T4> || onlyFirstIsVector<T4, T2, T3, T1>
+void makeBenchmarkTable(BenchmarkRecords* records, const bool overlap,
```

Comment:
Make a benchmark with ratio 1. That should be interessting.

Reply:
Done, see [acbd7c0](https://github.com/ad-freiburg/qlever/pull/860/commits/acbd7c04e936450cf77c825dc5246ae012e0a445).